src/daemon/dlt-daemon.c

```diff
@@ -3478,7 +3478,22 @@ int dlt_daemon_process_user_message_log(DltDaemon *daemon,
         /* Not enough bytes received to remove*/
         return DLT_DAEMON_ERROR_UNKNOWN;
 
+#ifdef DLT_SYSTEMD_WATCHDOG_ENABLE
+    const unsigned int start_time = dlt_uptime();
+#endif
+
     while (1) {
+#ifdef DLT_SYSTEMD_WATCHDOG_ENABLE
+        const unsigned int uptime = dlt_uptime();
+        if ((uptime - start_time) / 10000 > daemon->watchdog_trigger_interval) {
```

Comment:
when this situation occurs, the sd_notify is issued with every processed message. Increasing the system load much more.

I see two options here:
1) avoid looping over serveral messages here
2) update the start_time in case the sd_notify was triggered.

@alexmohr : what do you prefer?

Reply:
@michael-methner  in line 3493 the `while(1)` is canceled via `break`, meaning we don't call sd_notifiy each time. 
The closing braces belong to the `if` in line 3488. 
Therefore updating the start_time is superflous. Or am I missing something?

Reply:
@alexmohr : No, I was missing the break(); . THank you very much.